### DISCUSSION TITLE
create_network works but doesn't.

### DISCUSSION THREAD

**Discussion by evolvetechllc** [Art-of-WiFi/UniFi-API-client#234] [Q&A]
I'm able to connect successfully and for example list_networkconf works fine.  However when I do a create_network, I get back a response with the  with the site_id and _id, which presumably means the action worked, but when I go into the console the network doesn't show up.  Also, if I do another list_networkconf, the "new" network isn't listed.  Am I missing something? 

Edit:  I should mention this on a brand spanking new UDM Pro.

**Comment by malle-pietje** [comment]
Please share your exact code (but sanitized) here. That will allow us to figure out what is going on.

**Comment by evolvetechllc** [comment]
Here is my code....
```php
<?php

require_once('./UniFi-API-client/src/Client.php');

$unifi_connection = new UniFi_API\Client('username','password','https://{device ip}','default','8.4.62',false);
$login = $unifi_connection->login();

$network_to_add = array(
    "setting_preference" => "manual",
    "dhcpdv6_dns_auto" => 1,
    "ipv6_pd_stop" => "::7d1",
    "dhcpd_gateway_enabled" => null,
    "network_isolation_enabled" => 1,
    "ipv6_client_address_assignment" => "slaac",
    "dhcpd_start" => "10.99.9.50",
    "dhcpd_unifi_controller" => null,
    "ipv6_ra_enabled" => 1,
    "domain_name" => null,
    "ip_subnet" => "10.99.9.1/24",
    "ipv6_interface_type" => "none",
    "dhcpdv6_stop" => "::7d1",
    "is_nat" => 1,
    "dhcpd_dns_enabled" => null,
    "internet_access_enabled" => 1,
    "nat_outbound_ip_addresses" => array(),
    "dhcp_relay_enabled" => null,
    "dhcpd_conflict_checking" => 1,
    "ipv6_pd_auto_prefixid_enabled" => 1,
    "name" => "Subnet999",
    "site_id" => "{site id/random looking characters}",
    "dhcpdv6_leasetime" => 86400,
    "lte_lan_enabled" => 1,
    "dhcpd_leasetime" => 28800,
    "purpose" => "corporate",
    "igmp_snooping" => null,
    "dhcpd_time_offset_enabled" => null,
    "dhcpguard_enabled" => null,
    "dhcpdv6_allow_slaac" => 1,
    "ipv6_ra_preferred_lifetime" => 14400,
    "dhcpd_stop" => "10.99.9.254",
    "enabled" => 1,
    "dhcpd_enabled" => 1,
    "vlan" => 503,
    "dhcpd_wpad_url" => null,
    "networkgroup" => "LAN",
    "dhcpdv6_start" => "::2",
    "vlan_enabled" => 1,
    "ipv6_setting_preference" => "auto",
    "gateway_type" => "default",
    "ipv6_ra_priority" => "high",
    "dhcpd_boot_enabled" => null,
    "ipv6_pd_start" => "::2",
    "upnp_lan_enabled" => null,
    "dhcpd_ntp_enabled" => null,
    "mdns_enabled" => null,
    "dhcpd_tftp_server" => null,
    "auto_scale_enabled" => null
);


/*$network_to_add = $network_settings = [
    'name' => 'SampleNetwork',
    'vlan' => 900,
    'subnet' => '192.168.100.1/24',
    'dhcpd_start' => '192.168.100.10',
    'dhcpd_stop' => '192.168.100.250',
    'dhcpd_enabled' => true,
    'purpose' => 'corporate',
    'is_nat' => true,
    'enabled' => true
];*/

$x = $unifi_connection->create_network(json_encode($network_to_add));
if (!$x) { echo "Nope."; print_r($x); } else { print_r($x); }
$unifi_connection->logout();
```

```
Array
(
    [0] => stdClass Object
        (
            [site_id] => {site id matches what's in the code}
            [_id] => {new id of "random" characters}
        )

)
```

But when I log into the console, that new network is nowhere to be found.

**Reply by malle-pietje** [reply]
What is the output when you enable debug mode?

**Reply by evolvetechllc** [reply]
[debug.txt](https://github.com/user-attachments/files/17033817/debug.txt)
Here is the debug output.

**Reply by malle-pietje** [reply]
Okay, that output looks perfectly normal at first glance. It could be that there are inconsistencies in the setting values that prevent the controller form actually deploying the network.

What I typically do in such cases is pull the settings from an existing network into an array. Then remove the `_id` key, and change any settings as needed. You can probably also remove the `site_id` key because that should be added anyway when the controller creates the network.

Another thing I noticed is the high number of **null** values you're passing. The controller may not like that; here's what one of my test networks looks like where these keys either have a true or false value:
```json
    {
        "setting_preference": "manual",
        "dhcpdv6_dns_auto": true,
        "purpose": "corporate",
        "ipv6_pd_stop": "::7d1",
        "ipv6_client_address_assignment": "slaac",
        "dhcpd_start": "192.168.1.6",
        "ipv6_ra_preferred_lifetime": 14400,
        "ipv6_ra_enabled": true,
        "dhcpd_stop": "192.168.1.254",
        "domain_name": "localdomain",
        "dhcpd_enabled": true,
        "ip_subnet": "192.168.1.1/24",
        "ipv6_interface_type": "none",
        "networkgroup": "LAN",
        "dhcpdv6_start": "::2",
        "vlan_enabled": false,
        "ipv6_setting_preference": "auto",
        "dhcpdv6_stop": "::7d1",
        "is_nat": true,
        "dhcpdv6_enabled": false,
        "ipv6_ra_priority": "high",
        "dhcpd_conflict_checking": true,
        "ipv6_pd_start": "::2",
        "name": "Default",
        "site_id": "zzzzzzzzzzzzz",
        "dhcpdv6_leasetime": 86400,
        "mdns_enabled": false,
        "ipv6_enabled": true,
        "_id": "yyyyyyyyyyyyyyy",
        "attr_no_delete": true,
        "attr_hidden_id": "LAN",
        "lte_lan_enabled": true,
        "auto_scale_enabled": true
    }
```

**Reply by evolvetechllc** [reply]
That's exactly what I did.  I did a list_networkconf(), copy/pasted one of the networks I'd already setup, changed the values that needed to change and then fed that to 'create_network'.  I'll try it without all the nulls though and see if that helps.  I'd have thought they'd be fine since that's how the listing presented them.

**Reply by evolvetechllc** [reply]
Yah, that didn't fix it.  I even tried taking your example, changing the name, IPs and with and without site_id.  Still seemingly completes, but the network doesn't show up anywhere in the interface..... Until I went into the Legacy interface.  Then I see a BUNCH of extra networks with no settings.  So something definitely fishy somewhere.  I mean, I can add the networks manually, but this is being done to segregate tenants in a co-working space, so using the API would've kept everything consistent rather than having to double check everything 100x.
<img width="945" alt="networks" src="https://github.com/user-attachments/assets/42ee29e1-6d55-4e21-b6f5-2cb420c13494">
